unified_planning/model/walkers/type_checker.py

```diff
@@ -269,3 +270,22 @@ def walk_equals(
             ):
                 return None
         return BOOL
+
+    def walk_agent_exp(
```

Comment:
this method needs to be called `walk_dot` in order to work (or you need to use a `@handles` annotation like  for `walk_math_relation`)

Reply:
Also, this is a walker implementing a visitor pattern, so args will be a list of `Type` instances, not FNodes, and you can just return `args[0]`

Reply:
Done!